src/transformer.js

```diff
@@ -54,74 +89,111 @@ const transform = (file, api, options) => {
+    // Filter out the ones that are on the ignore list.
+    .filter(needsPathUpdate)
+    .replaceWith((item) => {
+      const replacement = updatePath(item, base);
+
+      // Replace the node with a new one on the AST.
+      return j.exportNamedDeclaration.from({
+        ...item.value,
+        source: j.literal(replacement),
+      });
+      // (
+      //   item.value.declaration,
+      //   item.value.specifiers,
+      //   j.literal(replacement),
+      // );
+    });
+  root
+    .find(j.ExportAllDeclaration)
+    // Filter out the ones that are on the ignore list.
+    .filter(needsPathUpdate)
+    .replaceWith((item) => {
+      const replacement = updatePath(item, base);
+
+      // Replace the node with a new one on the AST.
+      return j.exportAllDeclaration.from({
+        ...item.value,
+        source: j.literal(replacement),
+      });
+    });
   // =================================================
   // Parse the modules modifications.
   // =================================================
   if (cjs2esm.modules.length) {
+    const modules = cjs2esm.modules.map((r) => ({
+      ...r,
+      regex: new RegExp(`^${r.name}(/|$)`),
+    }));
+
     root
       .find(j.ImportDeclaration)
       // Filter out the import statments that don't need to be modified.
       .filter((item) => {
         const importPath = item.value.source.value;
-        return cjs2esm.modules.some((mod) => importPath.startsWith(mod.name));
+        return modules.some((mod) => mod.regex.test(importPath));
       })
       // Apply the modifications.
       .replaceWith((item) => {
         const importPath = item.value.source.value;
-        const info = cjs2esm.modules.find((mod) => importPath.startsWith(mod.name));
+        const info = modules.find((mod) => importPath.startsWith(mod.name));
         const find = info.find ? new RegExp(info.find) : new RegExp(`^${info.name}`);
         const replacement = importPath.replace(find, info.path);
 
         return j.importDeclaration(item.value.specifiers, j.literal(replacement));
       });
+    // This fails weirdly
+    // root
+    //   .find(j.exportNamedDeclaration)
+    //   // Filter out the export statments that don't need to be modified.
+    //   .filter((item) => {
+    //     const exportPath = item.value.source && item.value.source.value;
+    //     return exportPath && modules.some((mod) => mod.regex.test(exportPath));
+    //   })
+    //   // Apply the modifications.
+    //   .replaceWith((item) => {
+    //     const exportPath = item.value.source.value;
+    //     const info = cjs2esm.modules.find((mod) => exportPath.startsWith(mod.name));
+    //     const find = info.find ? new RegExp(info.find) : new RegExp(`^${info.name}`);
+    //     const replacement = exportPath.replace(find, info.path);
+
+    //     return j.exportNamedDeclaration.from({
+    //       ...item.value,
+    //       source: j.literal(replacement),
+    //     });
+    //   });
```

Comment:
Can you give me the expected input and output of the transformation that you are looking for in here?